lib/plug/router.ex

```diff
@@ -462,8 +462,16 @@ defmodule Plug.Router do
         conn = update_in unquote(conn).params, merge_params
         conn = update_in conn.path_params, merge_params
 
-        Plug.Conn.put_private(conn, :plug_route, {unquote(path), fn var!(conn) -> unquote(body) end})
+        Plug.Conn.put_private(conn, :plug_route, {append_match_path(conn, unquote(path)), fn var!(conn) -> unquote(body) end})
       end
+
+      defp append_match_path(%Plug.Conn{private: %{plug_route: {base_path, _}}}, path) do
```

Comment:
Defining this function inside `quote` means we will inject this same clause over and over for every route we define in the router. My suggestion is to move the whole put_private to a separate function in Plug.Router:

```elixir
@doc false
def __put_route__(conn, path, fun) do
  Plug.Conn.put_private(conn, :plug_route, {append_match_path(conn, path), fun})
end

defp append_match_route...
```

And then in the line above, instead of calling put_private plus append_match_path, we do:

```elixir
Plug.Router.__put_route__(conn, unquote(path), fn var!(conn) -> unquote(body) end)
```

Does it make sense?

Reply:
definitely! I only put it in the quote to be able to make it `defp`, but defining `__put_route__` outside make sense!

Reply:
(I've updated btw)